Riot/Modules/Common/Views/BadgedBarButtonItem.swift

```diff
@@ -0,0 +1,121 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+
+@objcMembers
+class BadgedBarButtonItem: UIBarButtonItem {
+    
+    var baseButton: UIButton
+    private var badgeLabel: UILabel
+    
+    private var theme: Theme = ThemeService.shared().theme
+    
+    var badgeText: String? {
+        didSet {
+            updateBadgeLabel()
+        }
+    }
+    var badgeBackgroundColor: UIColor {
+        didSet {
+            updateBadgeLabel()
+        }
+    }
+    var badgeTextColor: UIColor {
+        didSet {
+            updateBadgeLabel()
+        }
+    }
+    var badgeFont: UIFont {
+        didSet {
+            updateBadgeLabel()
+        }
+    }
+    var badgePadding: UIOffset {
+        didSet {
+            updateBadgeLabel()
+        }
+    }
+    
+    private var shouldHideBadge: Bool {
+        guard let text = badgeText else {
+            return true
+        }
+        return text.isEmpty || text == "0" || text == "nil" || text == "null"
+    }
+    
+    init(withBaseButton baseButton: UIButton) {
+        self.baseButton = baseButton
+        badgeBackgroundColor = .gray
+        badgeTextColor = .white
+        badgeFont = .systemFont(ofSize: 12, weight: .semibold)
+        badgePadding = UIOffset(horizontal: 8, vertical: 2)
+        badgeLabel = UILabel(frame: .zero)
+        badgeLabel.textAlignment = .center
+        badgeLabel.clipsToBounds = true
+        baseButton.addSubview(badgeLabel)
+        super.init()
```

Comment:
We can maybe use `super.init(customView: baseButton)` here

Reply:
We can't because it's a `convenience` initializer, but we need to call a `designated` initializer here.